### PR TITLE
feat: add openid connect auth

### PR DESCRIPTION
**Pull request by Fabriceli**
### Type of change:

- [x] New feature provided

### What this PR does / why we need it:

#1587 

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

**Comment by Fabriceli** [comment]
@tao12345666333 i add the e2e test for openid-connect auth for consumer, please rerun the CI, thanks.

**Comment by tao12345666333** [comment]
re-run all jobs

File: pkg/kube/apisix/apis/config/v2beta3/types.go [hunk 1]
```diff
@@ -338,11 +338,12 @@ type ApisixConsumerSpec struct {
 }
 
 type ApisixConsumerAuthParameter struct {
-	BasicAuth *ApisixConsumerBasicAuth `json:"basicAuth,omitempty" yaml:"basicAuth"`
-	KeyAuth   *ApisixConsumerKeyAuth   `json:"keyAuth,omitempty" yaml:"keyAuth"`
-	WolfRBAC  *ApisixConsumerWolfRBAC  `json:"wolfRBAC,omitempty" yaml:"wolfRBAC"`
-	JwtAuth   *ApisixConsumerJwtAuth   `json:"jwtAuth,omitempty" yaml:"jwtAuth"`
-	HMACAuth  *ApisixConsumerHMACAuth  `json:"hmacAuth,omitempty" yaml:"hmacAuth"`
+	BasicAuth     *ApisixConsumerBasicAuth     `json:"basicAuth,omitempty" yaml:"basicAuth"`
```
Comment (tao12345666333) [inline pkg/kube/apisix/apis/config/v2beta3/types.go:341]: v2beta3 has been deprecated and we don't need to add any functionality to it.

File: pkg/providers/apisix/translation/apisix_consumer.go [hunk 2]
```diff
@@ -57,6 +57,12 @@ func (t *translator) TranslateApisixConsumerV2beta3(ac *configv2beta3.ApisixCons
 			return nil, fmt.Errorf("invaild hmac auth config: %s", err)
 		}
 		plugins["hmac-auth"] = cfg
+	} else if ac.Spec.AuthParameter.OpenIDConnect != nil {
```
Comment (tao12345666333) [inline pkg/providers/apisix/translation/apisix_consumer.go:60]: ditto

**Review by tao12345666333 — COMMENTED** [review]
LGTM

v2beta3 has been deprecated, we don't need to change it.

**Comment by tao12345666333** [comment]
And please check the e2e job errors.

File: pkg/kube/apisix/apis/config/v2beta3/types.go (hunk 1 above)
Comment (Fabriceli) [reply pkg/kube/apisix/apis/config/v2beta3/types.go]: done

File: pkg/providers/apisix/translation/apisix_consumer.go (hunk 2 above)
Comment (Fabriceli) [reply pkg/providers/apisix/translation/apisix_consumer.go]: done

**Comment by Fabriceli** [comment]
> And please check the e2e job errors.

done

**Comment by Fabriceli** [comment]
As mention at https://github.com/apache/apisix/issues/8686, so close this PR

**Pull request closed without merging**